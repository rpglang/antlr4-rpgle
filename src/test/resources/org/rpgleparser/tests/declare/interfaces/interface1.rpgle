      DCL-PI DDR014;
        TRNSDT PACKED(7);
        TRNSDT2 LIKE(TRNSDT);
        dcl-parm TRNSDT3 LIKE(TRNSDT);
      END-PI;