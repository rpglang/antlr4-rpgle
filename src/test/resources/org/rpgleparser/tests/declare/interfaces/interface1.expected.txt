/*===TOKENS===*/
DS_ProcedureInterfaceStart<DCL-PI>
ID                       <DDR014>
FREE_SEMI                <;>
ID                       <TRNSDT>
PACKED                   <PACKED>
'('                      <(>
NUMBER                   <7>
')'                      <)>
FREE_SEMI                <;>
ID                       <TRNSDT2>
KEYWORD_LIKE             <LIKE>
'('                      <(>
ID                       <TRNSDT>
')'                      <)>
FREE_SEMI                <;>
DS_Parm                  <dcl-parm>
ID                       <TRNSDT3>
KEYWORD_LIKE             <LIKE>
'('                      <(>
ID                       <TRNSDT>
')'                      <)>
FREE_SEMI                <;>
DS_ProcedureInterfaceEnd <END-PI>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (dcl_pi
    DCL-PI
    (identifier (free_identifier (idOrKeyword DDR014)))
    ;
    (dcl_pi_field
      (identifier (free_identifier (idOrKeyword TRNSDT)))
      (datatype (datatypeName PACKED) (args ( (expression (number 7)) )))
      ;
    )
    (dcl_pi_field
      (identifier (free_identifier (idOrKeyword TRNSDT2)))
      (keyword_like
        LIKE
        (
        (simpleExpression (identifier (free_identifier (idOrKeyword TRNSDT))))
        )
      )
      ;
    )
    (dcl_pi_field
      dcl-parm
      (identifier (free_identifier (idOrKeyword TRNSDT3)))
      (keyword_like
        LIKE
        (
        (simpleExpression (identifier (free_identifier (idOrKeyword TRNSDT))))
        )
      )
      ;
    )
    (end_dcl_pi END-PI)
    ;
  )
)
/*======*/